### ISSUE TITLE
Properties from interfaces are not automapped

### ISSUE THREAD

**Issue by Bendetta** [thenerdery/UmbracoVault#29]
If I have a class with properties derived from an interface, those properties are not automapped.

```
public Interface INavItem
{
	string Name	{ get; set; }
}

[UmbracoEntity(Automap = true)]
public class BrokenViewModel : INavItem
{
	public string Name { get; set; }
}

[UmbracoEntity(Automap = true)]
public class WorkingViewModel : INavItem
{
	public string Name { get; set; }
	
	string INavItem.Name
	{
		get { return Name; }
		set { Name = value; }
	}
}
```

Doing `Vault.Context.GetContentById<BrokenViewModel>(id)` will result in Name being null, but `Vault.Context.GetContentById<WorkingViewModel>(id)` will fix it. Properties not defined in the interface still work as expected.

**Comment by kensykora** [comment]
Nice find.

**Comment by kensykora** [comment]
This is an issue specifically with the proxy instance factory that was a part of 1.2, the regular instance factory does not have this same issue.

**Comment by kensykora** [comment]
Fixed this in 1.2.2, pushed to internal nuget as 1.2.2-alpha1, let me know if this fixes your issue and I'll release it.

**Comment by NerderyMGlanzer** [comment]
How can I see the fix?  Really curious what the issue is.

On Fri, Apr 29, 2016 at 9:09 AM, Ken Sykora <notifications@github.com>
wrote:

> Closed #29 <https://github.com/thenerdery/UmbracoVault/issues/29>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/thenerdery/UmbracoVault/issues/29#event-645626805>
>



-- 

*Marshall Glanzer**Software Engineer and Co-President
<https://nerdery.com/copresident>**The Nerdery*

*marshall.glanzer@nerdery.com <marshall.glanzer@nerdery.com>*

**Comment by kensykora** [comment]
https://github.com/thenerdery/UmbracoVault/compare/master@%7B1day%7D...master#diff-1862e741114009b9bf91c3d97a14d60cR47
